app/models/registry.rb

```diff
@@ -0,0 +1,61 @@
+# model that represents a registry
+class Registry < ActiveRecord::Base
+  has_many :registry_mirrors, dependent: :destroy
+  has_one :certificate_service, as: :service, dependent: :destroy
+  has_one :certificate, through: :certificate_service
+
+  validates :name, presence: true, uniqueness: true
+  validates :url, presence: true, uniqueness: true, url: { schemes: ["https", "http"] }
+
+  class << self
+    # create or update suse Registry model
+    # rubocop:disable Metrics/CyclomaticComplexity,Metrics/PerceivedComplexity
+    def configure_suse_registry(suse_registry_mirror_params)
+      if suse_registry_mirror_params.blank?
+        cleanup
+        return []
+      end
+
+      errors      = []
+      cert        = suse_registry_mirror_params["certificate"]
+      mirror      = suse_registry_mirror_params["mirror"]
+      name        = suse_registry_mirror_params["name"] || "suse"
```

Comment:
I think SUSE in uppercase is better since it will be displayed in a few pages.

Reply:
Also, use a constant in `Registry` class and reuse everywhere instead of hard coded.

Reply:
done